### PR TITLE
read-only crud mode by hostname

### PR DESCRIPTION
**Pull request by peterbe**
Now the read-only is determined by the `window.location.hostname` NOT being in the default list of `localhost`, `localhost.org`, or `127.0.0.1`. 

To test this, let's pretend that `localhost.org` (or `127.0.0.1`) is the equivalent of `pr123.dev.content.mozit.cloud`. 
So type this:
```
export REACT_APP_CRUD_MODE_READONLY_HOSTNAMES=localhost
export REACT_APP_CRUD_MODE=true
yarn prepare-build && yarn start:server
```
Now it's like using the Yari preview server from the perspective of the mdn/content repo. 
To got http://localhost.org:5000/en-US/docs/Games/Tutorials#_flaws and http://localhost:5000/en-US/docs/Games/Tutorials#_flaws and compare. 
Or go to http://localhost.org:5000/en-US/_sitemap/Games vs. http://localhost:5000/en-US/_sitemap/Games

Man I'm so thankful for TypeScript for this! It made it super easy to refactor all that usage without allowing me to even start the server until all code made sense.

### CODE REVIEW

**Comment by peterbe** [comment]
Note! Once this lands we need to revisit our `.yml` files in that branch on mdn/content.

**Review by escattone — APPROVED** [review]
@peterbe This looks and works great, except I think we should consider changing the name of the constant.

File: client/src/constants.ts [hunk 1]
```diff
@@ -7,9 +7,13 @@ export const CRUD_MODE = JSON.parse(
     JSON.stringify(process.env.NODE_ENV === "development")
 );
 
-export const CRUD_MODE_READONLY = JSON.parse(
-  process.env.REACT_APP_CRUD_MODE_READONLY || "false"
-);
+export const CRUD_MODE_READONLY_HOSTNAMES = (
```
[the remark this answers is not in the document]
Comment (peterbe) [reply client/src/constants.ts:10]: I'm going to go for `CRUD_MODE_HOSTNAMES`

**Comment by peterbe** [comment]
Switched to `CRUD_MODE_HOSTNAMES`. I'm glad I wrote down the testing instructions because I could just retest and everything is still as expected.

**Pull request merged**